src/huggingface_hub/repository.py

```diff
@@ -643,13 +644,17 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
                 repo_url = repo_url.replace(f"{scheme}://", f"{scheme}://user:{token}@")
 
                 if namespace == user or namespace in valid_organisations:
-                    self.client.create_repo(
-                        repo_id=repo_id,
-                        token=token,
-                        repo_type=self.repo_type,
-                        exist_ok=True,
-                        private=self.private,
-                    )
+                    try:
+                        _ = HfApi().repo_info(f"{namespace}/{repo_id}")
+                    except HTTPError:
+                        self.client.create_repo(
```

Comment:
I'm very confused, why would `clone_from` create a repo remotely under any circumstances?

Reply:
It's because link given to `clone_from` might not be valid. Maybe it's better if I put a warning saying the remote repository doesn't exist instead. :)

Reply:
or do we want to not create at all and just raise a warning?

Reply:
I removed repo creation and added a warning.

Reply:
AH OK

Reply:
I just realized there are many tests that will fail and we'll break backwards compatibility if we change behavior of `clone_from`, I also thought that it was only used to clone existing repository when creating repository on local. pinging @LysandreJik about why the design is that way 😅

Reply:
(ok apparently you left a comment but I didn't see it bcs I didn't refresh the page lol)

Reply:
(i agree that Repository clone_from shouldn't create a repo)

Reply:
(can't find at the moment the past discussion with a potential explicit `Repository.create() -> Repository`)

Reply:
Yes, I also think that `clone_from` shouldn't create the repository if it doesn't exist, it isn't really its place to do so. I wouldn't say removing it is a bugfix as when I implemented it I shared it as such, but we can deprecate it for sure.

Reply:
I say we merge this and then deprecate it.

Reply:
SGTM!